syntax="proto2";

enum Enum {
  kInvalidEnum = 1;
  kEnum0 = 2;
}

message Bar {
  optional string nickname = 1 [default = "undefined-nickname"];
}

message Foo {
  optional string name = 1 [default="unnamed"];
  required int64 int_value = 10; 
  optional int64 opt_int_value = 11; 
  repeated int64 int_values = 12; 
  required string string_value = 50; 
  optional string opt_string_value = 51; 
  repeated string string_values = 52; 
  required Bar bar = 100;
  optional Bar optional_bar = 101; 
  repeated Bar bars = 102; 
  required Enum enum_value = 103; 
  optional Enum opt_enum_value = 104; 
  repeated Enum enum_values = 105; 
  map<int64, int64> map_int_int = 106;
  map<int64, int64> map_temp = 107;
  map<int64, Bar> map_bar = 108;
}
